In [3]:
import pandas as pd 
import numpy as np

In [4]:
df = pd.read_csv("D:\\João Carlos Moura\\brasileirao-data-analysis\\data\\raw\\brasileirao.csv")

df.head()

,ano_campeonato,data,rodada,estadio,arbitro,publico,publico_max,time_mandante,time_visitante,tecnico_mandante,...,chutes_bola_parada_mandante,chutes_bola_parada_visitante,defesas_mandante,defesas_visitante,impedimentos_mandante,impedimentos_visitante,chutes_mandante,chutes_visitante,chutes_fora_mandante,chutes_fora_visitante
0,2005,2005-10-08,31,Estádio Durival Britto e Silva,NaN,NaN,NaN,Paraná,Fluminense,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,2005,2005-04-23,1,Estádio Durival Britto e Silva,NaN,NaN,NaN,Paraná,Goiás EC,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,2005,2005-08-21,21,Estádio Durival Britto e Silva,NaN,NaN,NaN,Paraná,Vasco da Gama,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,2005,2005-08-28,23,Estádio Durival Britto e Silva,NaN,NaN,NaN,Paraná,São Paulo,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,2005,2005-06-26,9,Estádio Olímpico Nilton Santos,NaN,NaN,NaN,Botafogo,Figueirense FC,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [5]:
df.shape

(8453, 35)

In [6]:
df.columns

Index(['ano_campeonato', 'data', 'rodada', 'estadio', 'arbitro', 'publico',
       'publico_max', 'time_mandante', 'time_visitante', 'tecnico_mandante',
       'tecnico_visitante', 'colocacao_mandante', 'colocacao_visitante',
       'valor_equipe_titular_mandante', 'valor_equipe_titular_visitante',
       'idade_media_titular_mandante', 'idade_media_titular_visitante',
       'gols_mandante', 'gols_visitante', 'gols_1_tempo_mandante',
       'gols_1_tempo_visitante', 'escanteios_mandante', 'escanteios_visitante',
       'faltas_mandante', 'faltas_visitante', 'chutes_bola_parada_mandante',
       'chutes_bola_parada_visitante', 'defesas_mandante', 'defesas_visitante',
       'impedimentos_mandante', 'impedimentos_visitante', 'chutes_mandante',
       'chutes_visitante', 'chutes_fora_mandante', 'chutes_fora_visitante'],
      dtype='str')

In [7]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 8453 entries, 0 to 8452
Data columns (total 35 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   ano_campeonato                  8453 non-null   int64  
 1   data                            8453 non-null   str    
 2   rodada                          8453 non-null   int64  
 3   estadio                         8442 non-null   str    
 4   arbitro                         6743 non-null   str    
 5   publico                         6742 non-null   float64
 6   publico_max                     4181 non-null   float64
 7   time_mandante                   8453 non-null   str    
 8   time_visitante                  8453 non-null   str    
 9   tecnico_mandante                6299 non-null   str    
 10  tecnico_visitante               6299 non-null   str    
 11  colocacao_mandante              6743 non-null   float64
 12  colocacao_visitante             6743 non-null

In [8]:
df.describe


<bound method NDFrame.describe of       ano_campeonato        data  rodada  \
0               2005  2005-10-08      31   
1               2005  2005-04-23       1   
2               2005  2005-08-21      21   
3               2005  2005-08-28      23   
4               2005  2005-06-26       9   
...              ...         ...     ...   
8448            2017  2017-10-15      28   
8449            2017  2017-10-22      30   
8450            2017  2017-10-28      31   
8451            2017  2017-11-09      33   
8452            2017  2017-11-19      36   

                                          estadio                   arbitro  \
0                  Estádio Durival Britto e Silva                       NaN   
1                  Estádio Durival Britto e Silva                       NaN   
2                  Estádio Durival Britto e Silva                       NaN   
3                  Estádio Durival Britto e Silva                       NaN   
4                  Estádio Olímpico Nilton

In [9]:
df.isnull().sum()

ano_campeonato                       0
data                                 0
rodada                               0
estadio                             11
arbitro                           1710
publico                           1711
publico_max                       4272
time_mandante                        0
time_visitante                       0
tecnico_mandante                  2154
tecnico_visitante                 2154
colocacao_mandante                1710
colocacao_visitante               1710
valor_equipe_titular_mandante     2098
valor_equipe_titular_visitante    2098
idade_media_titular_mandante      2100
idade_media_titular_visitante     2100
gols_mandante                        1
gols_visitante                       1
gols_1_tempo_mandante             1721
gols_1_tempo_visitante            1721
escanteios_mandante               6366
escanteios_visitante              6366
faltas_mandante                   6366
faltas_visitante                  6366
chutes_bola_parada_mandan

In [10]:
df['ano_campeonato']

0       2005
1       2005
2       2005
3       2005
4       2005
        ... 
8448    2017
8449    2017
8450    2017
8451    2017
8452    2017
Name: ano_campeonato, Length: 8453, dtype: int64

In [11]:
df['arbitro']

0                            NaN
1                            NaN
2                            NaN
3                            NaN
4                            NaN
                  ...           
8448    Marcelo de Lima Henrique
8449                Rafael Traci
8450            Anderson Daronco
8451    Marcelo de Lima Henrique
8452              Wilton Sampaio
Name: arbitro, Length: 8453, dtype: str

In [12]:
df[['ano_campeonato', 'arbitro']]

,ano_campeonato,arbitro
0,2005,NaN
1,2005,NaN
2,2005,NaN
3,2005,NaN
4,2005,NaN
...,...,...
8448,2017,Marcelo de Lima Henrique
8449,2017,Rafael Traci
8450,2017,Anderson Daronco
8451,2017,Marcelo de Lima Henrique


## Observações iniciais

- A base contém **8453 partidas** e **35 colunas**.
- Os dados abrangem diversas temporadas do Campeonato Brasileiro.
- Algumas colunas apresentam valores ausentes, principalmente **arbitro**, indicando informações não registradas em determinadas partidas.
- Não foram encontradas linhas duplicadas.
- A coluna **data** foi convertida para o formato datetime.
- As estatísticas indicam que os gols e o público possuem valores coerentes, sem números negativos.
- A base contém informações detalhadas, incluindo estatísticas ofensivas, defensivas, escanteios, faltas, chutes e dados dos elencos.
- A riqueza das variáveis permitirá realizar análises descritivas, comparações entre equipes e, futuramente, modelos preditivos.